src/plugins/data_views/server/has_user_index_pattern.ts

```diff
@@ -15,19 +19,31 @@ interface Deps {
   soClient: SavedObjectsClientContract;
 }
 
-export const hasUserIndexPattern = async ({ esClient, soClient }: Deps): Promise<boolean> => {
-  const indexPatterns = await soClient.find<IndexPatternSavedObjectAttrs>({
+export const hasIndexPattern = async ({
+  esClient,
+  soClient,
+}: Deps): Promise<SavedObjectsFindResponse<IndexPatternSavedObjectAttrs, unknown>> =>
+  soClient.find<IndexPatternSavedObjectAttrs>({
     type: 'index-pattern',
     fields: ['title'],
     search: `*`,
     searchFields: ['title'],
     perPage: 100,
   });
 
+export const hasUserIndexPattern = async (
+  { esClient, soClient }: Deps,
+  iPatterns?: SavedObjectsFindResponse<IndexPatternSavedObjectAttrs, unknown>
```

Comment:
Do we ever use this function without passing `iPatterns` ? if not: I suggest to make it required to make it simpler